webapp/src/modules/mortgage/selectors.js

```diff
@@ -0,0 +1,32 @@
+import { createSelector } from 'reselect'
+
+import { FETCH_ACTIVE_PARCEL_MORTGAGES_REQUEST } from './actions'
+import { isLoadingType } from 'modules/loading/selectors'
+import { getAddress } from 'modules/wallet/selectors'
+import { getParcels } from 'modules/parcels/selectors'
+import { getActiveMortgagesByBorrower } from 'modules/mortgage/utils'
+import { buildCoordinate } from 'lib/utils'
+
+export const getState = state => state.mortgages
+export const getData = state => getState(state).data
+export const getLoading = state => getState(state).loading
+export const getMortgages = state => getData(state).mortgages
+export const isFetchingParcelMortgages = state =>
+  isLoadingType(getLoading(state), FETCH_ACTIVE_PARCEL_MORTGAGES_REQUEST)
+export const getMappedMortgages = state =>
+  Object.keys(getMortgages(state)).map(key => getMortgages(state)[key])
```

Comment:
I don't know if `Mapped` is the right word here (I had to check the implementation to get what it was doing). Maybe `getMortgagesArray`? (I know it's not _that_ different)

Reply:
☝️ @nachomazzara @NicoSantangelo **watch out**, this is missing `reselect`! 

A new instance of the array is created every time the selector is called. 

This is used by another selector `getMortgagedParcels` in `modules/parcels/selectors`, and this will kill that selectors memoization too.

`getMortgagedParcels` is used by `getAddresses` which is used by `getWallet`(both wont be able to memoize their results).

`getWallet` is used by 13 containers, and their `mapStateToProps` gets called on every `dispatch`. Since the selectors they use are not memoized, the `shallowCompare` they do will return false, triggering renders down the React tree (not only the containers, also their children if they receive the non-memoized prop).